File: LeetCode/Array/Python/Leetocde1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        ans,n,i = [],len(s),0
        while i < n:
            j,cnt = i + 1,1
            while j < n and cnt != 0:
                if s[j] == '(': cnt += 1
                else: cnt -= 1
                j += 1
            ans.append(s[i + 1:j - 1])
            i = j
        return ''.join(ans)

File: LeetCode/Array/JS/Leetocde1021-remove-outermost-parentheses.js
var removeOuterParentheses = function (s) {
    let [ans, n] = [[], s.length];
    for (let i = 0; i < n; i++) {
        let [j, cnt] = [i + 1, 1];
        while (j < n && cnt != 0) {
            if (s[j] == '(') cnt++;
            else cnt--;
            j++
        }
        ans.push(s.slice(i + 1, j - 1));
        i = j - 1;
    }
    return ans.join('');
};